docs/internals/layout_in_memory.rst

```diff
@@ -5,36 +5,126 @@
 Layout in Memory
 ****************
 
+Reserved Memory Areas
+=====================
+
 Solidity reserves four 32-byte slots, with specific byte ranges (inclusive of endpoints) being used as follows:
 
 - ``0x00`` - ``0x3f`` (64 bytes): scratch space for hashing methods
-- ``0x40`` - ``0x5f`` (32 bytes): currently allocated memory size (aka. free memory pointer)
+- ``0x40`` - ``0x5f`` (32 bytes): allocated memory size (aka. free memory pointer)
 - ``0x60`` - ``0x7f`` (32 bytes): zero slot
 
-Scratch space can be used between statements (i.e. within inline assembly). The zero slot
-is used as initial value for dynamic memory arrays and should never be written to
-(the free memory pointer points to ``0x80`` initially).
+Memory Allocation
+-----------------
+
+Solidity is a high level language that is translated (compiled) into EVM instructions. Its features
```

Comment:
This section is about internals of the compiler. I think having to read two paragraphs of explanation what allocation is distracts from the useful information in this section.